DEVELOPERS.md

```diff
@@ -320,3 +318,77 @@ In particular, the SPDX header may look like:
+#define RZ_NULLABLE  /* pointer can be null */
+#define RZ_DEPRECATE /* should not be used in new code and should/will be removed in the future */
+```
+
+Most of them are easy to understand and you can see brief explanation in the comments. But `RZ_OWN` and `RZ_BORROW` may be a little tricky to new developers.
+
+Sometimes it may not be immediately clear whether the object you are getting from a function shall be freed or not. Rizin uses `RZ_OWN` and `RZ_BORROW` to indicate pointer ownership so you don't have to read complicated function definitions to know whether they should still free objects or not.
+
+You can use the two modifiers in two places and their explanations are as below:
+
+- before the return type of function
+  - `RZ_OWN`: the ownership of the returned object is transferred to the caller. The caller *owns* the object, so it must free it (or ensure that something else frees it).
+  - `RZ_BORROW`: the ownership of the returned object is not transferred. The caller can use the object, but it does not own it, so it should not free it.
+- before the parameter of function
+  - `RZ_OWN`: the ownership of the passed argument is transferred to the callee. The callee now owns the object and it is its duty to free it (or ensure that something else frees it). In any case, the caller should not care anymore about freeing that passed object.
+  - `RZ_BORROW`: the ownership of the passed argument is *not* transferred to the callee, which can use it but it should not free it. After calling this function, the caller still owns the passed object and it should ensure that at some point it is freed.
+
+Examples:
+
+```c
+RZ_OWN MyString *capitalize_str(RZ_BORROW char *s) {
+  MyString *m = RZ_NEWS(MyString);
+  m->s = strdup(s);
+  capitalize(m->s);
+  return m;
+}
+
+int main() {
+  char *s = strdup("Hello World");
+  MyString *m = capitalize_str(s);
+  // s was RZ_BORROW, so main still need to free it
+  free(s);
+  // ... use m ....
+  // m was RZ_OWN, so main now has to free it
+  my_string_free(m);
+}
+```
+
+```c
+RZ_BORROW MyString *capitalize_str(RZ_BORROW MyFile *f, RZ_OWN char *s) {
+  MyString *m = RZ_NEWS(MyString);
+  m->s = s;
+  capitalize(m->s);
+  f->m = m;
+  return m;
+}
+
+int main() {
+  char *s = strdup("Hello World");
+  MyFile *f = create_my_file();
+  MyString *m = capitalize_str(f, s);
+  // s was RZ_OWN, so main does not need to free it. s is now owned by `m`
+  // ... use m ....
+  // m was RZ_BORROW, so main is just borrowing it from `f`, and it does not have to free it.
+  my_file_free(f);
+  // f was created by main and never transferred to anything else, so main needs to free it.
+}
+```
+
+- You should use these modifiers consistently in both function definition and declaration.
+- We should use these modifiers when and only when it makes sense. For example, if your function returns `const char *`, the caller should not free it because of the `const`. So specifying `RZ_BORROW` in this case is probably redundant.
```

Comment:
```suggestion
- You should use these modifiers when and only when it makes sense. For example, if your function returns `const char *`, the caller should not free it because of the `const`. So specifying `RZ_BORROW` in this case is probably redundant.
```
For consistency with the sentence above